### PR TITLE
azurerm_netapp_pool - support for the encryption_type property

### PR DESCRIPTION
**Pull request by Orexii**
This PR adds support for double encryption for ANF capacity pools and corresponding tests and documentation. Below are the results of the tests:

```
user [ ~/go/src/github.com/terraform-providers/terraform-provider-azurerm ]$ make acctests SERVICE='netapp' TESTARGS=' -parallel 5 -run=TestAccNetAppPool_' TESTTIMEOUT='1200m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/netapp -parallel 5 -run=TestAccNetAppPool_ -timeout 1200m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNetAppPool_basic
=== PAUSE TestAccNetAppPool_basic
=== RUN   TestAccNetAppPool_requiresImport
=== PAUSE TestAccNetAppPool_requiresImport
=== RUN   TestAccNetAppPool_complete
=== PAUSE TestAccNetAppPool_complete
=== RUN   TestAccNetAppPool_update
=== PAUSE TestAccNetAppPool_update
=== RUN   TestAccNetAppPool_doubleEncryption
=== PAUSE TestAccNetAppPool_doubleEncryption
=== CONT  TestAccNetAppPool_basic
=== CONT  TestAccNetAppPool_update
=== CONT  TestAccNetAppPool_requiresImport
=== CONT  TestAccNetAppPool_doubleEncryption
=== CONT  TestAccNetAppPool_complete
--- PASS: TestAccNetAppPool_doubleEncryption (208.13s)
--- PASS: TestAccNetAppPool_basic (212.30s)
--- PASS: TestAccNetAppPool_requiresImport (216.54s)
--- PASS: TestAccNetAppPool_complete (219.55s)
--- PASS: TestAccNetAppPool_update (359.96s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp        (cached)
user [ ~/go/src/github.com/terraform-providers/terraform-provider-azurerm ]$ make acctests SERVICE='netapp' TESTARGS=' -parallel 5 -run=TestAccDataSourceNetAppPool_' TESTTIMEOUT='1200m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/netapp -parallel 5 -run=TestAccDataSourceNetAppPool_ -timeout 1200m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceNetAppPool_basic
=== PAUSE TestAccDataSourceNetAppPool_basic
=== CONT  TestAccDataSourceNetAppPool_basic
--- PASS: TestAccDataSourceNetAppPool_basic (207.55s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp        207.559s
user [ ~/go/src/github.com/terraform-providers/terraform-provider-azurerm ]$
```

### CODE REVIEW

File: internal/services/netapp/netapp_pool_resource.go [hunk 1]
```diff
@@ -88,6 +89,17 @@ func resourceNetAppPool() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"encryption_type": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Computed: true,
```
Comment (paulomarquesc) [inline internal/services/netapp/netapp_pool_resource.go:95]: This should be removed and include `Default` instead, set default to Single.

File: website/docs/r/netapp_pool.html.markdown [hunk 2]
```diff
@@ -46,14 +46,16 @@ The following arguments are supported:
...
-* `service_level` - (Required) The service level of the file system. Valid values include `Premium`, `Standard`, or `Ultra`. Changing this forces a new resource to be created.
+* `service_level` - (Required) The service level of the file system. Valid values include `Premium`, `Standard`, and `Ultra`. Changing this forces a new resource to be created.
 
 * `size_in_tb` - (Required) Provisioned size of the pool in TB. Value must be between `2` and `500`.
 
 ~> **NOTE** `2` TB capacity pool sizing is currently in preview. You can only take advantage of the `2` TB minimum if all the volumes in the capacity pool are using `Standard` network features. If any volume is using `Basic` network features, the minimum size is `4` TB. Please see the product [documentation](https://learn.microsoft.com/azure/azure-netapp-files/azure-netapp-files-set-up-capacity-pool) for more information.
 
 * `qos_type` - (Optional) QoS Type of the pool. Valid values include `Auto` or `Manual`.
 
+* `encryption_type` - (Optional) The encryption type of the pool. Valid values include `Single`, and `Double`. Changing this forces a new resource to be created.
```
Comment (paulomarquesc) [inline website/docs/r/netapp_pool.html.markdown:57]: With my early suggestion, let's add that it defaults to `Single`

**Review by paulomarquesc — COMMENTED** [review]
LGTM, did some minor suggestions.

File: website/docs/r/netapp_pool.html.markdown (hunk 2 above)
Comment (Orexii) [reply website/docs/r/netapp_pool.html.markdown]: fixed

**Review by katbyte — APPROVED** [review]
LGTM 📺

**Pull request merged**